src/containers/EmailVerification.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import apiClient from "../lib/apiClient";
+
+import ButtonLink from "../components/ButtonLink";
+
+export default class EmailVerification extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      confirmationError: ""
+    };
+  }
+
+  componentDidMount = () => {
+    const request = {
+      verification_token: this.props.match.params.emailConfirmationId
+    };
+
+    apiClient("PUT", "/signup", request).catch(err => {
+      err.text().then(errorMsg => {
+        this.setState({
+          confirmationError: errorMsg
+        });
+      });
+    });
+  };
+
+  render() {
+    return (
+      <div className="pre-confirmation">
+        <h1>Email Verification</h1>
+        {this.state.confirmationError && (
+          <div>
+            <h2>Sorry, this email confirmation code can't be validated.</h2>
```

Comment:
Not a big deal, but can you escape the ' or just wrap the text with a template literal so github doesn't add this formatting stuff?

Reply:
done.